pkg/server/webhook.go

```diff
@@ -164,20 +167,63 @@ func (whsvr *WebhookServer) getSidecarConfigurationRequested(ignoredList []strin
 		return "", ErrSkipAlreadyInjected
 	}
 
-	// determine whether to perform mutation based on annotation for the target resource
-	requestedInjection, ok := labels[requestAnnotationKey]
-	if !ok {
-		glog.Infof("Pod %s/%s annotation %s is missing, skipping injection", metadata.Namespace, metadata.Name, requestAnnotationKey)
-		return "", ErrMissingRequestAnnotation
-	}
-	injectionKey := strings.ToLower(requestedInjection)
-	if !whsvr.Config.HasInjectionConfig(requestedInjection) {
-		glog.Errorf("Mutation policy for pod %s/%s: requested injection %s was not in configuration, skipping", metadata.Namespace, metadata.Name, requestedInjection)
-		return requestedInjection, ErrRequestedSidecarNotFound
+	// Check if the configuration is all pods
+	// Check if this is a pod of one of the selected deployments
+	// Last option is to check the label (backwards compatibility)
```

Comment:
This comment belongs next to or inside `checkIfShouldInject` function.

Reply:
it was comment for me. I removed it

Reply:
It was a good comment.